test/tstLinearBVH.cpp

```diff
@@ -26,24 +26,16 @@
 
 namespace tt = boost::test_tools;
 
-template <typename T>
-struct TreeTypeTraits;
-
-template <typename... DeviceTypes>
-struct TreeTypeTraits<std::tuple<DeviceTypes...>>
+BOOST_AUTO_TEST_CASE_TEMPLATE(empty_tree, DeviceType, ARBORX_DEVICE_TYPES)
 {
-  using type = std::tuple<ArborX::BVH<DeviceTypes>...>;
-};
-
-using TreeTypes = typename TreeTypeTraits<ARBORX_DEVICE_TYPES>::type;
+  using Tree = ArborX::BVH<typename DeviceType::memory_space>;
+  using ExecutionSpace = typename DeviceType::execution_space;
 
-BOOST_AUTO_TEST_CASE_TEMPLATE(empty_tree, Tree, TreeTypes)
-{
-  using device_type = typename Tree::device_type;
   // tree is empty, it has no leaves.
   for (auto const &tree : {
-           Tree{},         // default constructed
-           make<Tree>({}), // constructed with empty view of boxes
+           Tree{}, // default constructed
+           makeTree<ExecutionSpace, Tree>(
```

Comment:
FTAD should kick in if you reorder template params.

Reply:
Don't get it.

Reply:
You will only need to write `make<Tree>(exec_space, ...)`